### ISSUE TITLE
env variable for API token

### ISSUE THREAD

**Issue by Yvonne-Aizawa** [civitai/comfy-nodes#19]
When i use this node and i share my workflow it would mean that my civit api token will be sent with it. 

i would like either an option in the settings menu or the option to set an env variable.

**Comment by maximilianwicen** [comment]
Indeed, this is a crazy security flaw and I'm a bit baffled that the creator of this addon did not think about this...

**Comment by WASasquatch** [comment]
It's not really a "security flaw". That would simply be user error, in reality. Just delete the key, then save the workflow safe for sharing using the build-in "Save" feature for user-curated exports. 

There isn't a convention for the API key as a system var in the documentation, so I was hesitant to just pick a name. As [according to the documentation](https://education.civitai.com/civitais-guide-to-downloading-via-api/) the expected usage is just trailing URLs or header info. That in itself is a security flaw since it doesn't guarantee it from unwanted eyes if DNSSEC of not setup or user is compromised with malware.

The key is also simply used to authorize download of locked models at this time, so I'm not sure it warrants any over the top complaints. I maintain this repo out of my spare time.  

> i would like either an option in the settings menu or the option to set an env variable.

An option for using a env var is certainly doable. I guess a generic key variable name could `CIVITAI_TOKEN`

**Comment by lrojas94** [comment]
> It's not really a "security flaw". That would simply be user error, in reality. Just delete the key, then save the workflow safe for sharing using the build-in "Save" feature for user-curated exports.
> 
> There isn't a convention for the API key as a system var in the documentation, so I was hesitant to just pick a name. As [according to the documentation](https://education.civitai.com/civitais-guide-to-downloading-via-api/) the expected usage is just trailing URLs or header info. That in itself is a security flaw since it doesn't guarantee it from unwanted eyes if DNSSEC of not setup or user is compromised with malware.
> 
> The key is also simply used to authorize download of locked models at this time, so I'm not sure it warrants any over the top complaints. I maintain this repo out of my spare time.
> 
> > i would like either an option in the settings menu or the option to set an env variable.
> 
> An option for using a env var is certainly doable. I guess a generic key variable name could `CIVITAI_TOKEN`

Actually, the API Key has quite a few more uses (Training, generation, etc). So ideally it should not be exposed, as people might end up using other people's funds.

**Comment by ecker00** [comment]
This saves a lot of clicking around to fill it in every-time, and improves security. I think both options can be available still, so we don't break comparability. But if no key is given in the node, look for an ENV variable would solve it.

I suggest `CIVITAI_COMFY_NODES_API_KEY` is clear and useful for the powerusers. 

Then mention this behavior in the readme so people can take advantage of the env variable.
